### DISCUSSION TITLE
Unable to resolve DT::datatable crossref

### DISCUSSION THREAD

**Discussion by rohitfarmer** [quarto-dev/quarto-cli#3034] [Q&A]
Hi there, I am building a Quarto blog and getting errors upon cross-referencing a `DT::datatable`. Cross-referencing works with figures, section labels, and `kable()` tables, but not with `DT::datatable`. Below is the code that generates a tibble and displays with datatables which, upon rendering says `WARNING: Unable to resolve crossref @tbl-rdatasets` and I get `?@tbl-rdatasets` where I cite this table. 

````
---
title: "Testing datatable"
author: 
  - name: "Rohit Farmer"
    orcid: "0000-0003-4197-3047"
date: "2022-10-25"
format:
  html:
    code-fold: true
    code-tools:
      source: false
      toggle: true
---

See @tbl-rdatasets


```{r}
#| label: tbl-rdatasets
#| tbl-cap: "Default datasets in R"

suppressMessages(library(DT))
suppressMessages(library(tidyverse))
dat <- data()
dat <- as_tibble(dat$results) %>% dplyr::select(-LibPath) %>%
  dplyr::filter(Package == "datasets")
DT::datatable(dat)
```
````

Any suggestions would be helpful. Thanks!

**Comment by mcanouil** [comment]
I am AFK right now, but is it the code you are really using?
It seems like an error, you prefixed an R object with the name of a package.
Could you provide a small working but complete Quarto document that exhibits the error?

**Reply by rohitfarmer** [reply]
I have updated the post with a complete Quarto document that is throwing the error. Thanks.

**Reply by mcanouil** [reply]
This feels like a bug ... on which I can't say what's the source ...
Maybe it has something to do with the fact that `DT` is using the [DataTable.js library](https://datatables.net/) instead of regular HTML as `knitr::kable` and `gt`.

Below a fully reproducible example:

````markdown
---
format: html
---

See @tbl-rdatasets1 and @tbl-rdatasets2

```{r}
#| label: tbl-rdatasets1
#| tbl-cap: "Default datasets in R"
# library(DT, quietly = TRUE)
DT::datatable(head(mtcars))
```

```{r}
#| label: tbl-rdatasets2
#| tbl-cap: "Default datasets in R"
# library(knitr, quietly = TRUE)
knitr::kable(head(mtcars))
```
````

**Reply by cscheid** [reply]
Yes, I'd characterize it a missing feature on quarto.

We make an effort to detect which libraries are used to generate html entries and how, but in general that is a best-effort solution, and it seems that we don't detect `DT::datatable` well enough. We're in the process of designing an API for third-party libraries so they can advertise their output to quarto (so we don't need to detect every table printing library on github and CRAN), but it will be a few months before it lands.

**Reply by mcanouil** [reply]
Is there an issue to track this already? (I did not find one)

**Reply by cscheid** [reply]
It's spread all over the `table` and `crossref`-labeled issues, so we know what it's about.

**Reply by mcanouil** [reply]
Thanks 😊

**Reply by rohitfarmer** [reply]
Thank you, @mcanouil and @cscheid, for answering my question. It seems like cross-referencing DT tables is not an option right now. I  will wait for the update. I am new to Quarto; therefore, I was scared I might be doing something wrong.
